crypto/asn1/evp_asn1.c

```diff
@@ -95,21 +116,64 @@ int ASN1_TYPE_get_int_octetstring(const ASN1_TYPE *a, long *num,
     if (atmp == NULL)
         goto err;
 
-    if (num != NULL)
-        *num = atmp->num;
+    ret = asn1_type_get_int_oct(atmp->oct, atmp->num, num, data, max_len);
 
-    ret = ASN1_STRING_length(atmp->oct);
-    if (max_len > ret)
-        n = ret;
-    else
-        n = max_len;
-
-    if (data != NULL)
-        memcpy(data, ASN1_STRING_get0_data(atmp->oct), n);
     if (ret == -1) {
  err:
         ASN1err(ASN1_F_ASN1_TYPE_GET_INT_OCTETSTRING, ASN1_R_DATA_IS_WRONG);
     }
     M_ASN1_free_of(atmp, asn1_int_oct);
     return ret;
 }
+
+typedef struct {
+    ASN1_OCTET_STRING *oct;
+    int32_t num;
+} asn1_oct_int;
+
+ASN1_SEQUENCE(asn1_oct_int) = {
+        ASN1_SIMPLE(asn1_oct_int, oct, ASN1_OCTET_STRING),
+        ASN1_EMBED(asn1_oct_int, num, INT32)
+} static_ASN1_SEQUENCE_END(asn1_oct_int)
+
+DECLARE_ASN1_ITEM(asn1_oct_int)
+
+int ASN1_TYPE_set_octetstring_int(ASN1_TYPE *a, long num, unsigned char *data,
+                                  int len)
+{
+    asn1_oct_int atmp;
+    ASN1_OCTET_STRING oct;
+
+    atmp.num = num;
+    atmp.oct = &oct;
+    asn1_type_init_oct(&oct, data, len);
+
+    if (ASN1_TYPE_pack_sequence(ASN1_ITEM_rptr(asn1_oct_int), &atmp, &a))
+        return 1;
+    return 0;
+}
+
+int ASN1_TYPE_get_octetstring_int(const ASN1_TYPE *a, long *num,
```

Comment:
Its not clear to me why these two functions should sit in the generic ASN.1 sub-library. They don't seem especially useful except for a very specific purpose. Certainly I don't see them as public functions.

Reply:
They are almost the same like `ASN1_TYPE_get_int_octetstring` resp. `ASN1_TYPE_get_int_octetstring` except the order in ASN.1 sequence. The use is quite similar as well so I put them to the same file. Would you prefer a different file? The reason why they are public is for consistency with  `ASN1_TYPE_get_int_octetstring` resp. `ASN1_TYPE_get_int_octetstring` that are also public but happy to make them private if you think it makes more sense?

Reply:
Ok, keep them where they are, but please make them private.

Reply:
Done